Domains step uses TwoColumnLayout

p-jackson opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/wp-calypso/issues/101199

## Proposed Changes

* Wraps the domain step in a `<Step.TwoColumnLayout>` wrapper when the `onboarding/step-container-v2` flag is present.
* Adds `AsyncDomainStepWrapper` to async load the components which will be used to present the step.
* Uses the `<Step.TwoColumnLayout>` to lay out the column elements on mobile
  * Previously the sidebar content was duplicated in the main content column, and media queries where used to show/hide the appropriate elements
  * Now there's only one copy of the sidebar content and the layout changes from a flex row to a flex column.

There are a number of custom additions to the floating toolbar. For that reason this step isn't yet using the `<Step.StickyBottomBar>` component.

The responsiveness of the component is a bit broken. There are some browser widths where it really does not look good. And it doesn't automatically switching to the mobile layout, it needs a page refresh. This PR doesn't address these issues, they're in the production version. Though I would consider fixing these as part of the consistency work.

## Testing Instructions


https://github.com/user-attachments/assets/e4d2fbd1-9c86-4325-a199-5970581cd4d0


https://github.com/user-attachments/assets/ef5dfebf-4fcd-4c87-9dc0-bf3bc2f03cf4

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Test the step at `/setup/onboarding`
* Double check tracks events
* Ensure you also test using `?flags=-onboarding/step-container-v2` too to confirm that when the flag isn't present, nothing has changed ([excluding the sticky footer padding changes](https://github.com/Automattic/wp-calypso/pull/101431#discussion_r2006853962))

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

p-jackson committed:
Unified domain step using the new TwoColumnLayout wireframe

zaguiini reviewed (commented):
I know this is just a draft, but here are my testing results so far:

### Theirs

![image](https://github.com/user-attachments/assets/8862b790-a178-4bb4-9e45-423b5f691fbf)

### Ours

![image](https://github.com/user-attachments/assets/c2738c14-a474-4a44-b824-5b7cf410e9ab)

<hr />

There are differences in the input, and the back button has blue text 🤔 [This is what is causing it](https://github.com/Automattic/wp-calypso/blob/update/domain-step-3-column-right/client/landing/stepper/declarative-flow/internals/global.scss#L46). One more customization we can remove when it's all done 👍🏻 

We're also missing the heading sub text there.

I'm seeing the "Already own a domain?" section twice, both in the desktop and mobile version. Also, on mobile, I think the address bar image should always stretch so it follows the main column's width.

![image](https://github.com/user-attachments/assets/a2c6c54c-9dd5-4293-b79b-a27024225641)

When I type and select a domain, it looks broken:

| Ours | Theirs |
| ----- | ------ |
| ![image](https://github.com/user-attachments/assets/a3fc3e9f-5030-4940-bf44-fd8e570f54ae) | ![image](https://github.com/user-attachments/assets/72846c35-649a-42ea-9e81-073c37531e3f) |

Finally, there's no bottom bar in our version, but I guess we're still going to do it with a custom `StickyBottomBarComponent` 😁 👍🏻 

| Ours | Theirs |
| ---- | ----- |
| ![image](https://github.com/user-attachments/assets/22d24425-928b-4e84-a715-64560f4e4b04) | ![image](https://github.com/user-attachments/assets/c130006f-bb5b-415e-bec4-b2801abc2300) |

p-jackson committed:
Fix back button color

p-jackson committed:
Fixup padding of WordPress wordmark

p-jackson committed:
Remove duplicate sidebar content

The new step wireframe will be responsible for the responsive layout, no
need to duplicate DOM elements.

p-jackson committed:
Fix styling of domains step in step container v2

p-jackson committed:
Fix layout on mobile

p-jackson committed:
Remove unused classes

p-jackson commented on client/signup/steps/domains/async-domain-step-wrapper.tsx:
```diff
@@ -0,0 +1,51 @@
+	className: string;
+}
+
+export default function AsyncDomainStepWrapper( props: AsyncDomainStepWrapperProps ) {
```
Do we even need this async wrapper component? I created it because the other step wrapper was being async loaded too. But I'm not sure it even has any affect. The big components get loaded when `renderContent()` gets called, so I'm not sure how much difference these small components make.

zaguiini replied:
This component is also used in Start, right?

If yes, then we do need it, as the other instances of StepContainerV2 are only used within stepper and the steps are also loaded async.

p-jackson commented on client/signup/steps/domains/style.scss:
```diff
@@ -84,7 +84,7 @@ body.is-section-stepper .domains__step-content,
 		left: 0;
 		z-index: 99;
-		padding-bottom: 6px !important;
+		padding-bottom: 0 !important;
```
Note that this change is being made unconditionally, regardless of the feature flag. That's because this padding looks off in the production version too.

p-jackson commented on client/signup/steps/domains/style.scss:
```diff
@@ -423,6 +400,10 @@ body.is-section-signup {
 			border-bottom: none;
 		}
 
+		.domains__domain-cart-foldable-card .domains__domain-side-content {
```
Same here, this change is being made unconditionally, regardless of the feature flag. That's because this padding looks off in the production version too.

p-jackson commented on packages/onboarding/src/step-container-v2/components/buttons/BackButton/style.scss:
```diff
@@ -1,5 +1,6 @@
 .step-container-v2__back-button {
 	--wp-components-color-accent: var(--color-neutral-100);
+	--color-link: var(--color-neutral-100);
```
The WP component library says that links are supposed to be blue. So having black links in the onboarding flow really is a special case.

p-jackson commented on packages/onboarding/src/step-container-v2/wireframes/TwoColumnLayout/TwoColumnLayout.tsx:
```diff
@@ -42,14 +42,7 @@ export const TwoColumnLayout = ( props: TwoColumnLayoutProps ) => {
-							{ child }
-						</div>
-					);
+					return <div style={ { flex: getChildFlexGrow( index ) } }>{ child }</div>;
```
Removing unused class

zaguiini committed:
Reset some styles to improve the responsiveness of some elements within the step

zaguiini committed:
Make sure the side content is hidden within the main container
